### ISSUE TITLE
Error when formatting syntax related to `for condition where method() {}`

### ISSUE THREAD

**Issue by aksswami** [nicklockwood/SwiftFormat#136]
I am using swift format to format my iOS project and for two instances swift format is not formatting correctly.

```swift
      let dataModels = ApplicationProgressManager.shared.currentDataModal(formType)
         var isValid: Bool = true
-        for dataModel in dataModels where !dataModel.isValid() {   // before formatting
+        for dataModel in dataModels where !dataModel.isValid {    // after formatting
             isValid = false
         }
```

```swift
-                    for error in errors where (error["errorType"] as? String)?.lowercased() == "XYZ".lowercased() {
+                    for error in errors where (error["errorType"] as? String)?.lowercased() == "XYZ".lowercased {
```

Is this expected behaviour or some there is an issue with swiftFormat.

**Comment by nicklockwood** [comment]
@aksswami this looks like a bug. I'll look into it - thanks!

**Comment by aksswami** [comment]
Another example:- https://github.com/hyperoslo/Sugar/blob/master/Source/Shared/Extensions/Array+Queueable.swift

If you run swiftformat on above it will change to:-

`for (index, element) in currentQueue.enumerated() where element.process {`

**Comment by nicklockwood** [comment]
Thanks for the PR! Fixed in 0.28.5.
